Common/VR/PPSSPPVR.cpp

```diff
@@ -420,20 +442,147 @@ bool StartVRRender() {
 
 	if (VR_InitFrame(VR_GetEngine())) {
 
+		// Get OpenXR view and fov
+		XrFovf fov = {};
+		XrPosef invViewTransform[2];
+		for (int eye = 0; eye < ovrMaxNumEyes; eye++) {
+			XrView view = VR_GetView(eye);
+			fov.angleLeft += view.fov.angleLeft / 2.0f;
+			fov.angleRight += view.fov.angleRight / 2.0f;
+			fov.angleUp += view.fov.angleUp / 2.0f;
+			fov.angleDown += view.fov.angleDown / 2.0f;
+			invViewTransform[eye] = view.pose;
+		}
+
+		// Get 6DoF scale
+		float scale = 1.0f;
+		bool hasUnitScale = false;
+		if (PSP_CoreParameter().compat.vrCompat().UnitsPerMeter > 0) {
+			scale = PSP_CoreParameter().compat.vrCompat().UnitsPerMeter;
+			hasUnitScale = true;
+		}
+
+		// Update matrices
+		for (int matrix = 0; matrix < VR_MATRIX_COUNT; matrix++) {
+			if (matrix == VR_PROJECTION_MATRIX) {
+				float nearZ = g_Config.fFieldOfViewPercentage / 200.0f;
+				float tanAngleLeft = tanf(fov.angleLeft);
+				float tanAngleRight = tanf(fov.angleRight);
+				float tanAngleDown = tanf(fov.angleDown);
+				float tanAngleUp = tanf(fov.angleUp);
+
+				float M[16] = {};
```

Comment:
`Lin::Matrix4x4` has a method that would do this and then you could simply do `M.wx *= -1.0f;` etc. before the memcpy().  Just noting.

-[Unknown]

Reply:
Thanks, I see there method for ortho but not for perspective. I will keep it as it is as I would rearrange the values so much that it wouldn't make the code cleaner.